lib/application_insights/rack/track_request.rb

```diff
@@ -54,7 +58,11 @@ def call(env)
           :url => request.url
         }
 
-        @client.track_request id, start_time, duration, status, success, options
+        # Setup operation context
+        @client.context.operation.id = operation_id(request_id)
+        @client.context.operation.parent_id = request_id
```

Comment:
parent_id should be env['Http_Request_Id']. Here the request_id is the id of the request telemetry itself.